src/core/service/notifications/registry/handler.go

```diff
@@ -148,6 +154,9 @@ func (n *NotificationHandler) Post() {
 			}
 		}
 		if action == "pull" {
+			// publish image pull event
+			buildAndPublishImagePullOrPushEvent(topic.PullImageTopic, pro, event)
```

Comment:
Again we better separate the build event and publish, and use a factory to build the event.

Reply:
> Again we better separate the build event and publish, and use a factory to build the event.

fixed, separate build and publish， use Factory Pattern to build event